### ISSUE TITLE
FileNotFoundError: [Errno 2] No such file or directory: 'outputs/traj/test_0_pX0_traj.pdb'

### ISSUE THREAD

**Issue by bethleegy** [RosettaCommons/RFdiffusion#206]
Hi there,

I am trying to diffuse a binder of length 20 to the X chain of the protein "2O26." 

<img width="346" alt="image" src="https://github.com/RosettaCommons/RFdiffusion/assets/17113877/b3aa2649-cbd9-4de5-8481-1b1c1884e7c4">

But I can't seem to get around this error. If this is an issue within the PDB file, I would appreciate some pointers in how to start identifying it. 

Thank you in advance.

**Comment by TheFoodborneScientist** [comment]
I was experiencing the same issue on the Google Colab version, but it appears the issue has been resolved as of March 5th 12:30 CST. I would recommend trying again!

**Comment by bethleegy** [comment]
@TheFoodborneScientist thank you, it seems some proteins work now, but not the one I posted about. Weird! I'll keep looking into it.

**Comment by TheFoodborneScientist** [comment]
It seems like I spoke too soon! I'm having issues again with the notebook. Perhaps it's intermittent as well?

**Comment by Gianni1202** [comment]
Yes I am having the same problem, inability to generate an output, any protein I use isn't working

**Comment by nd346** [comment]
Hi I am having the same issue, does anyone knows the answer?

**Comment by itzfredz** [comment]
![Screen Shot 2024-03-09 at 14 34 10](https://github.com/RosettaCommons/RFdiffusion/assets/132261351/e456fafc-1a1c-44ed-b3bd-1cb455fb0012)

Same here, the issue is still up.

**Comment by sokrypton** [comment]
Do you guys have an example? I'll take a look.

**Comment by itzfredz** [comment]
> Do you guys have an example? I'll take a look.

Hi there, unsure what to give as an example but the error occurs after I inputted all the necessary details for the contigs and such in the "run RFdiffusion to generate a backbone" tab. This then showed the upload prompt (since Ileft my pdb field blank as I wanted to upload my own structures). And then once the .pdb is fully uploaded it shows that error and RFdiffusion stops entirely. Hope that helps! Let me know if you need any more details! :)

**Comment by patjiang** [comment]
Hello, I've been running into this one as well. I traced it back a bit and it seems that the error comes from the internal notebook's function defined as "run_diffusion"

**Comment by patjiang** [comment]
From my previous comment, it seems that the eroor is this section of code:

`for n in range(num_designs):
    pdbs = [f"outputs/{path}/{path}_{n}_pX0_traj.pdb",
            f"outputs/{path}/{path}_{n}_Xt-1_traj.pdb",
            f"{full_path}_{n}.pdb"]
    for pdb in pdbs:
      with open(pdb,"r") as handle: pdb_str = handle.read()
      with open(pdb,"w") as handle: handle.write(fix_pdb(pdb_str, contigs))`

**Comment by patjiang** [comment]
Hello all. I was able to resolve the issue by directly modifying my .pdb file to only contian the atomic information, as well as inputting my .pdb file into the RFDiffusion/examples/input_pdbs folder.

Furthermore, the issue returned when I passed arguments into the hotspots text box. I am unsure what is causing this issue, but I hope @sokrypton can bring me some clarity.

**Comment by sokrypton** [comment]
If you are seeing this error that means RfDiffusion failed to run. It could fail for a number of reasons including:
- protein too large to run, ran out of memory
- inputs don't make sense (ex. chains specified for range or hotspot don't match what is found in the pdb)

**Comment by xinran1129305** [comment]
> If you are seeing this error that means RfDiffusion failed to run. It could fail for a number of reasons including:
> 
> * protein too large to run, ran out of memory
> * inputs don't make sense (ex. chains specified for range or hotspot don't match what is found in the pdb)

I just have tried to get a protein of about 25aa, and I have the same input as before. But it seems that we would get this error now, which is never meet before. @sokrypton Thanks in advance!

**Comment by xinran1129305** [comment]
Oh, new error occurs when I try to setup RFdiffusion: ImportError: Cannot load Graphbolt C++ library

**Comment by patjiang** [comment]
@xinran1129305 make sure you have gpu connected

**Comment by patjiang** [comment]
@xinran1129305 I have gotten the colab notebook to work fine for me; could you please show an image of your inputs so we can help you debug?

**Comment by liszt-c** [comment]
This error is caused by a failed RFdiffusion run without output. I tried some methods to solve this problem on my local computer. If you still get this error after carefully checking the input files and installation steps, you may try the following actions.

Code files under RFdiffusion\inference have parts of circular references (imports) that can cause errors. You may need to use the importlib module to handle code with circular references. In colab, the run_command_and_get_pid() function in the setup RFdiffusion section uses pid to detect the running condition of RFdiffusion. It may fail to run due to permission problems, so you can comment out the function. Run the command using os.system(f'{cmd}') directly from the # RUN comment.

In addition, there is some dependency problem, but I do not think it is the key cause of this problem. jax[cuda12] relies on CUDA12.3 while pytorch and dgl rely on CUDA12.1, you may need to reinstall the available jax, which usually causes the colabdesign module to not call the GPU and only work with the cpu. dgl relies on pytorch2.2.1 and later, and the latest version of pytorch2.2.2. Is installed by default, but this part generally does not cause conflicts.

I'm new to github, and forgive me for my broken English.

**Comment by reynunfra** [comment]
Hello everyone,
I wanted to share a solution that has consistently worked for me when encountering this error in the Colab notebook.
After experiencing this issue multiple times, I've found that completely resetting the runtime environment fixes the problem:

Go to "Runtime" > "Disconnect and delete runtime" in Colab
Reconnect and re-execute the RFdiffusion setup from the beginning

This has worked reliably for me in all my recent attempts. I believe there might be some residual files or states that persist between runs, causing conflicts with new executions. A clean restart seems to clear these out.

Hope this helps others who are facing the same issue! :)
